Cabal/Distribution/Simple.hs

```diff
@@ -292,6 +294,12 @@ cleanAction hooks flags args = do
     pbi <- preClean hooks args flags'
 
     (_, ppd) <- confPkgDescr hooks verbosity
+    -- It might seem like we are doing something clever here
+    -- but we're really not: if you look at the implementation
+    -- of 'clean' in the end all the package description is
+    -- used for is to clear out @extra-tmp-files@.  IMO,
+    -- the configure script goo should go into @dist@ too!
+    --          -- ezyang
```

Comment:
You know, you'd be shocked how often people ask me questions that could be solved by blame ;)